mne/io/base.py

```diff
@@ -1629,6 +1629,36 @@ def append(self, raws, preload=None):
                 len(self._raw_extras) == len(self._filenames)):
             raise RuntimeError('Append error')  # should never happen
 
+    def append_channels(self, raw_list):
+        """Append new channels to the Raw instance.
+
+        Parameters
+        ----------
+        raw_list : list
+            A list of Raw objects to append to self
+
+        Returns
+        -------
+        new_raw : RawArray
+            A Raw object with new channels appended
+        """
+        from .array import RawArray
+        if not isinstance(raw_list, (list, tuple)):
+            raise AssertionError('Input must be a list or tuple of Raw objs')
+
+        if not all([raw.preload for raw in raw_list]):
+            raise AssertionError('All data must be preloaded')
+
+        if not all(self.n_times == raw.n_times for raw in raw_list):
+            raise AssertionError('Data have unequal numbers of times')
+
+        # Pull data from each raw object to be appended, then vstack
+        infos, datas = zip(*[(raw.info, raw._data) for raw in raw_list])
+        data = np.vstack([self._data] + list(datas))
+        new_info = _merge_info([self.info] + list(infos))
+        new_raw = RawArray(data, new_info)
+        return new_raw
```

Comment:
should mofify inplace, think about sustainable naming compatible with epochs and evoked for future mixin